### ISSUE TITLE
[Crop & Soil] Allow users to have more control over whether synthetic fertilizers are used

### ISSUE THREAD

**Issue by ew3361zh** [RuminantFarmSystems/RuFaS#1828]
#### Problem
In Field._execute_manure_application(), if there is unmet nitrogen/phosphorus demand after the manure application, the demand is attempted to be met by 1 of 2 synthetic fertilizer mixes:

1. An ONLY_NITROGEN_MIX if the unmet_phosphorus_demand == 0.0
2. A more optimized mix that attempts to apply the least amount of fertilizer based on the demanded nutrients.

These synthetic fertilizers are applied whether or not the user wants them and regardless of cost.

#### What needs to happen:
There should be a way for the user to set whether they want synthetic fertilizer applied at all to meet nutrient demand (as opposed to applying purchased manure perhaps - issue #1827). Perhaps this could be regulated by a certain cost/amount per application. And we eventually would like to be tracking the cost of the applied synthetic fertilizer though that likely fits better within the EEE module.

**Comment by KFosterReed** [comment]
I thought that the input `supplement_manure_nutrient_dificiencies`, which can be true or false did give the user the option to not apply additional fertilizer in the cases where the manure supply does not meet the nutrient demand?

**Comment by KFosterReed** [comment]
I would rank this as medium priority and fairly feasible. Within the context of how manure requests and applications work right now, the `supplement_manure_nutrient_dificiencies` input could be changed from a boolean value to a string that indicates whether it is synthetic fertilizer or manure (so options would be something like `"manure", "synthetic fertilizer", "None"`

**Comment by allisterakun** [comment]
This is the sister issue of #1827.
@Isgotkowitz will come back and explain the current state of manure application specifications in the CS module. 
The dev team member who takes on this task should follow up with the SMEs to confirm future plans.

**Comment by Isgotkowitz** [comment]
@KFosterReed is correct, `supplement_manure_nutrient_deficiencies` will control whether or not nutrient deficient manure applications are supplemented with synthetic fertilizer. 

This input is a `Field` level configuration i.e. one field can have additional synthetic fertilizer applied while another does not, but _all_ manure applications to a field will either be supplemented with fertilizer or not.

Question for SMEs: would it be desirable to have the ability for one manure application to be supplemented with synthetic fertilizer, one to be supplemented with manure, and one that isn't supplemented at all?

**Comment by matthew7838** [comment]
> @KFosterReed is correct, `supplement_manure_nutrient_deficiencies` will control whether or not nutrient deficient manure applications are supplemented with synthetic fertilizer.
> 
> This input is a `Field` level configuration i.e. one field can have additional synthetic fertilizer applied while another does not, but _all_ manure applications to a field will either be supplemented with fertilizer or not.
> 
> Question for SMEs: would it be desirable to have the ability for one manure application to be supplemented with synthetic fertilizer, one to be supplemented with manure, and one that isn't supplemented at all?

@adhikaripradip75 Will take another look into this question.

**Comment by adhikaripradip75** [comment]
Yes, in the situation all the required crop nutrient is supplied by the manure, synthetic fertilizer is applied to supply the deficient portion of nutrient with the available fertilizer in the market. In practical dairy farmers apply manure and fertilizer based on the nutrient concentration of the manure, soil test NPK results and crop yield goals. In many cases dairy farmers use manure to supply P and use synthetic fertilizer to supply deficient portion of N.
